packetbeat/protos/amqp/amqp_fields.go

```diff
@@ -0,0 +1,199 @@
+package amqp
+
+import (
+	"encoding/binary"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func getTable(fields common.MapStr, data []byte, offset uint32) (next uint32, err bool, exists bool) {
+	ret := make(common.MapStr)
+
+	length := binary.BigEndian.Uint32(data[offset : offset+4])
+
+	// size declared too big
+	if length > uint32(len(data[offset+4:])) {
+		return 0, true, false
+	}
+	if length > 0 {
+		exists = true
+		err := fieldUnmarshal(ret, data[offset+4:offset+4+length], 0, length, -1)
+		if err {
+			logp.Warn("Error while parsing a field table")
+			return 0, true, false
+		}
+		if fields == nil {
+			fields = ret
+		} else {
+			fields.Update(ret)
+		}
+	}
+	return length + 4 + offset, false, exists
+}
+
+func getArray(fields common.MapStr, data []byte, offset uint32) (next uint32, err bool, exists bool) {
+	ret := make(common.MapStr)
+
+	length := binary.BigEndian.Uint32(data[offset : offset+4])
+
+	// size declared too big
+	if length > uint32(len(data[offset+4:])) {
+		return 0, true, false
+	}
+	if length > 0 {
+		exists = true
+		err := fieldUnmarshal(ret, data[offset+4:offset+4+length], 0, length, 0)
+		if err {
+			logp.Warn("Error while parsing a field array")
+			return 0, true, false
+		}
+		if fields == nil {
+			fields = ret
+		} else {
+			fields.Update(ret)
+		}
+	}
+	return length + 4 + offset, false, exists
+}
+
+//The index parameter, when set at -1, indicates that the entry is a field table.
+//If it's set at 0, it is an array.
+func fieldUnmarshal(table common.MapStr, data []byte, offset uint32, length uint32, index int) (err bool) {
```

Comment:
It depends of where the table is. If the table is right in the middle of a frame, and thus followed by other parameters, stream must be dropped because the offset is not right anymore and you don't know where are the additional arguments when you get out of the function. If it's at the end of a frame (for example in queueDeclareMethod), stream is not dropped but the Notes field is updated to indicates that the parsing of the table failed.